### PR TITLE
feat:added backup phrase for new users

### PR DESCRIPTION
**Pull request by Bibitha-V**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,6 +59,31 @@ let icon = Path.join(__dirname, 'assets/icon.png')
...
+
+function checkAndCreateMnenonicKeys(cb) {
+  
+  const secretFile= Path.join(u.dataLoc(), '__ssb','secret')
+  if(fs.existsSync(secretFile)) {
+    return cb()
+  }else{
+    openewUserWindow() 
+    electron.ipcMain.on('main-window',  (ev, winData) => {
+      startMainWindow()
```
Comment (theproductiveprogrammer) [inline index.js]: This should call cb() again

File: index.js [hunk 2]
```diff
@@ -147,7 +176,41 @@ electron.app.on('ready', () => {
...
+  },
+   title: 'Everlife Explorer',
+    show: true,
+    titleBarStyle: 'hiddenInset',
+    autoHideMenuBar: true,
+    backgroundColor: '#EEE',
+    icon: icon 
+    });
+    userWindow.loadURL(
+    'file://' + Path.join(__dirname, '.', 'newuser','1.html')
```
Comment (theproductiveprogrammer) [inline index.js]: Why is this newuser/1.html? The path looks a little strange

Comment (theproductiveprogrammer) [reply index.js]: Maybe newuser/step-1, step-2 etc would be better

File: newuser/newuser.js [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+var phrase='';
+var selectrdPhraseArr=[];
+var pubkey;
+var seckey;
+var publickey;
+var secretkey;
+var elifeKeys;
+var secureKeys = '';
+var passMatch= false;
+if (document.URL.split('?')[0].includes('4.html')){
```
Comment (theproductiveprogrammer) [inline newuser/newuser.js]: This should probably be invoked from newuser/4.html instead

i.e. in newsier/4.html script tag add a call to a function `generateKey(document.getElementById('generate'))`  or something similar

File: newuser/newuser.js [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    const params = new URLSearchParams(window.location.search)
+    for (const param of params) {
+        pharseParam=param[1].split('^^')[0];
+        pharseParam2 = param[1].split('^^')[1];
+        
+    }
+    
+    document.getElementById('generate').innerHTML = pharseParam;
```
Comment (theproductiveprogrammer) [inline newuser/newuser.js]: Why is this being set and then overwritten on line 36?

File: newuser/newuser.js [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
...C7u5/41oB1HmBfFAAkQvUUwQtUeQjRa6zzYGZZLAH5xob++paWr4+PjV+865HLH7fOA0yHAoDEcVyvRdLJkj4k6TTrPNi/LJSAfH39nvq21kslXdi/6bJ11nmA2aIAINGqZeBESe8XKwOxlMYSkG9s7K9vWXCZpEsY+kgqCgBSo/ocgqMlvUNcMxAraSgBhQXNj9Y1NV2ssbHv9D98OfvxI/EoAEil6rMIni3pdEkfkbTEOlPWJbEE1C9suyXf0HDx2NDQdezFj7ShACATHNfrlPR8SW+X9GbrPFkV9xJQ19S0tdDc9H3l898e3LzlHm7ZQ5pRAJA51dWBAyW9RNI7JZ1knSlL4lQC8vX1Q4WWBb/MFwrfGh0aumGgfNUu60xAVCgAyLzqswkOknSKKhsQvcg6U9pZlYB8ff1goWXB9flC4fvjo6PX9z98+TbrzwKwQgEA9uG4XkHSMyQdr8o1BGdaZ0qjKEpAobn5ibqm4i9zdXU/HRsZuWXXw5f3Wh83EBcUAGAGjutJ0iJJz1HllsMzVNmHAPMUZAnINzbsLjQ335qvL1wj5fzhvr41e7Z8b9T6GIG4ogAANahuU9wh6VmSjlNlQyL2IahBLSWgrrlpe12xeGu+UPi1crkbxwYHH9r16JWD1scCJAkFAAhIdaWgWdIKSQdLOkaVlYJTrbPF3f5KQF1TU19dsfHefH3hJuXz/6Vx3T28c+cTe7Z+f8w6M5B0FAAgZNViUFRlxeAAVU4lHKbKNQYnSSpYZzQ0IulmSbf3b9hYGB8dfSiXy909Pj6+ZnhH7/Y9W7/HbXhASCgAgLHqRYcLVLnOYLkqJeGZqmxkdIikQ1UpD0mzTdL9kh6UtEbSekmPSHpc0nZJu8p+acQ6JJBV/x9iXVBA6uYbKQAAACV0RVh0ZGF0ZTpjcmVhdGUAMjAyMS0wMy0wN1QwNjo0MzowOCswMDowMI3/JdUAAAAldEVYdGRhdGU6bW9kaWZ5ADIwMjEtMDMtMDdUMDY6NDM6MDgrMDA6MDD8op1pAAAAY3RFWHRzdmc6Y29tbWVudAAgR2VuZXJhdG9yOiBBZG9iZSBJbGx1c3RyYXRvciAxOS4wLjAsIFNWRyBFeHBvcnQgUGx1Zy1JbiAuIFNWRyBWZXJzaW9uOiA2LjAwIEJ1aWxkIDApICDOSJALAAAAAElFTkSuQmCC')
```
Comment (theproductiveprogrammer) [inline newuser/newuser.js]: Can't we save the images as files and load them as normal images? The inline-data for images makes it harder to read the code

**Review by theproductiveprogrammer — CHANGES_REQUESTED** [review]
Have made some review comments. Nicely done Bibitha.

File: newuser/newuser.js (hunk 5 above)
Comment (Bibitha-V) [reply newuser/newuser.js]: updated as review comment

File: index.js (hunk 2 above)
Comment (Bibitha-V) [reply index.js]: updated as per review comment.

File: index.js (hunk 1 above)
Comment (Bibitha-V) [reply index.js]: updated as review comment

**Review by Bibitha-V — COMMENTED** [review]
updated

File: newuser/newuser.js (hunk 3 above)
Comment (Bibitha-V) [reply newuser/newuser.js]: I tried the same but, upon redirection  the previous variable value set is getting cleared on undefined.

File: newuser/newuser.js (hunk 4 above)
Comment (Bibitha-V) [reply newuser/newuser.js]: newly constructed HTML string  I am pushing in to the existing DOM.

File: newuser/newuser.js (hunk 3 above)
Comment (theproductiveprogrammer) [reply newuser/newuser.js]: Variables will always get cleared on new pages. That's to be expected. Can you detail the issue a bit more? What variable and how can this solve that problem?

File: newuser/newuser.js (hunk 4 above)
Comment (theproductiveprogrammer) [reply newuser/newuser.js]: You are writing two things to the DOM. The first here and the second a few lines later which will overwrite this. So what's the use of writing something only to overwrite it after?

Comment (Bibitha-V) [reply newuser/newuser.js]: I wrote that line for testing initially, thanks for pointing ,I have removed that line .

File: newuser/newuser.js (hunk 3 above)
Comment (Bibitha-V) [reply newuser/newuser.js]: Updated as Function

**Comment by theproductiveprogrammer** [comment]
Excellent job 👍

**Pull request merged**